README.md

```diff
@@ -1,9 +1,53 @@
+> In the case of the MCUboot example, the _"factory firmware"_ is saved and would require manual transfer when the board is indeed connected. In either case, the demonstration step would be the only part that requires manual intervention from the user.
 >
-> Please note that the documentation for these examples will be migrated soon to this repository (and may see some changes). 
+> **Important**: For now, the tool assumes the target board, toolchain, and example unless otherwise specified by the end-user. Currently, the only board and toolchain supported are `NRF52840_DK` and `GCC_ARM` respectively.
+
+## Examples
+
+Please refer to the example-specific READMEs for build and demonstration instructions.
+1. [Mock Example](Mock)
+2. [MCUboot Example](MCUboot)
+
+## Known Issues
+There are slight kludges with how the MCUboot and Mock examples are built with `fota.sh`. The former is a direct result of dependency conflicts between [mbed-tools](https://github.com/ARMmbed/mbed-tools), [pyocd](https://github.com/pyocd/pyOCD), and [mbed-os](https://github.com/ARMmbed/mbed-os). The latter is a result of issues with flashing the target binary to the connected board. The details of both are addressed below:
+
+1. **Dependency Conflicts**: The difference in version requirements of the [PyYAML](https://pyyaml.org) dependency imposed by both mbed-os and pyocd led to the creation of a temporary virtual environment; this is used in the "_creating and flashing the factory firmware_" stage of the MCUboot example build process. This is a known issue and would require changes to the `requirements.txt` file in the mbed-os to resolve.\
+   \
+   Another minor conflict that doesn't hinder the build process is one between mbed-tools and mbed-os on the version requirement of the [Click](https://click.palletsprojects.com/en/8.0.x/) dependency; mbed-tools requires that the minimum version of Click to be greater than 7.1 while mbed-os requires it to be greater than 7.0. Now, the dependencies for mbed-os are installed after those of mbed-tools, which overwrites the newer version and generates a conflict. This would (again) require changes to the requirements file in the mbed-os repository to bump up the minimum version number of Click to 7.1.
+
+2. **Target Binary Flashing**: For the Mock example, compiling with mbed-tools results in the following error:
+    ```
+    ERROR: Build program file (firmware) not found <path to mock example>/target/cmake_build/<target board>/develop/<toolchain>/target.hex
+    ``` 
+   The tool is looking for a hex file under the cmake build output whose name comes from project directory (in this case, "target"). However, the generated one is named `BLE_GattServer_FOTAService.hex`. Again, this is a known issue and has been [filed](https://github.com/ARMmbed/mbed-tools/issues/282) (282) in the mbed-tools repository by [noonfom](https://github.com/noonfom).
```

Comment:
```suggestion
   The tool is looking for a hex file under the cmake build output whose name comes from project directory (in this case, "target"). However, the generated one is named `BLE_GattServer_FOTAService.hex`.
```

Reply:
Is this even relevant? Does your script not already take this into account?

Reply:
Yes, you are correct that it does take it into account - the script copies the binary to the mount point (if one's provided). I was initially stuck wondering why the `--flash` option with `mbed-tools` failed to work. I thought that it would be useful, potentially time-saving information for an end-user in the case that they either modify the script (to include --flash) or decide be adventurous and build manually. 

However, I now see your point that it does look a bit irrelevant, since the aim to simplify the process of building and demonstrating the example - _"it should just work"_ and not hinder an end-user's first experience. We could go ahead and remove this section entirely since the example-specific scripts have notes mentioning these - lines 57-59 and 100-102 of `mcuboot.sh` and lines 35-37 of `mock.sh`.